tempest/api/workloadmgr/barbican/test_barbican.py

```diff
@@ -2833,3 +2833,142 @@ def test_8_barbican(self):
+            #create another key pair...
+            kp1 = self.create_key_pair(tvaultconf.key_pair_name)
+
+            #create another vm...
+            vm_id1 = self.create_vm(key_pair=kp1)
+
+            #find volume_type = luks. So that existing encrypted volume type can be used.
+            #Get the volume_type_id
+            vol_type_id = -1
+            for vol in CONF.volume.volume_types:
+                if (vol.lower().find("luks") != -1):
+                    vol_type_id = CONF.volume.volume_types[vol]
+
+            if (vol_type_id == -1):
+                reporting.add_test_step("No encrypted volume available to attach", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+                raise Exception("No luks volume found to create encrypted volume.")
+
+            #Now create volume with derived volume type id...
+            volume_id1 = self.create_volume(
+                volume_type_id=vol_type_id)
+
+            #Attach volume to vm...
+            self.attach_volume(volume_id1, vm_id1)
+
+            #now edit the workload with workload_modify cli and try to attach vm with encrypted volume...
+            try:
+                workload_modify_command = command_argument_string.workload_modify + "--instance instance-id=" + \
+                    str(vm_id1) + " --instance instance-id=" + str(vm_id) + " " + str(wid)
+
+                rc = cli_parser.cli_returncode(workload_modify_command)
+                if rc != 0:
+                    LOG.error("Command did not execute correctly")
+                    reporting.add_test_step(
+                        "Failed to execute workload-modify command to add one more vm",
+                        tvaultconf.PASS)
+                    reporting.set_test_script_status(tvaultconf.PASS)
+                else:
+                    LOG.debug("Command executed correctly")
+
+                    self.wait_for_workload_tobe_available(wid)
+                    #check for vm count - If it is more than 1 then test case failed...
+                    workload_vm_count = query_data.get_available_vms_of_workload(wid)
+
+                    if (workload_vm_count == 2):
+                        reporting.add_test_step(
+                            "Unencrypted Workload connected with another instance with encrypted volume",
+                            tvaultconf.FAIL)
+                        reporting.set_test_script_status(tvaultconf.FAIL)
+                        raise Exception("Added encrypted vm to existing workload. Failed to run test case")
+                    else:
+                        reporting.add_test_step(
+                            "After workload modify vm count is 1. Unencrypted Workload cannot have instance with encrypted volume.",
+                            tvaultconf.PASS)
+                        reporting.set_test_script_status(tvaultconf.PASS)
+
+            except Exception as e:
```

Comment:
This exception block does not get called here and error message verification gets skipped. Error msg verification need to be handled.